app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/client/RtcClientMessageController.kt

```diff
@@ -0,0 +1,104 @@
+package co.netguru.baby.monitor.client.feature.communication.webrtc.client
+
+import co.netguru.baby.monitor.client.feature.communication.webrtc.IceCandidateState
+import co.netguru.baby.monitor.client.feature.communication.webrtc.OnIceCandidateAdded
+import co.netguru.baby.monitor.client.feature.communication.webrtc.base.RtcMessageHandler
+import co.netguru.baby.monitor.client.feature.communication.websocket.Message
+import co.netguru.baby.monitor.client.feature.communication.websocket.RxWebSocketClient
+import com.google.gson.Gson
+import com.google.gson.JsonParseException
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.plusAssign
+import io.reactivex.rxkotlin.subscribeBy
+import org.webrtc.IceCandidate
+import org.webrtc.SessionDescription
+import timber.log.Timber
+import java.net.URI
+
+class RtcClientMessageController(
+    private val gson: Gson,
+    private val serverUri: URI,
+    private val rxWebSocketClient: RxWebSocketClient
+) {
+
+    lateinit var rtcMessageHandler: RtcMessageHandler
+    private var compositeDisposable = CompositeDisposable()
+
+    fun startRtcSession(
+        sessionDescription: SessionDescription
+    ) {
+
+        if (rxWebSocketClient.isOpen()) sendOffer(sessionDescription)
+        compositeDisposable += rxWebSocketClient.events(serverUri = serverUri)
+            .doOnNext {
+                if (it is RxWebSocketClient.Event.Open)
+                    sendOffer(sessionDescription)
+            }
+            .ofType(RxWebSocketClient.Event.Message::class.java)
+            .subscribe { event: RxWebSocketClient.Event.Message ->
+                handleIncomingMessage(event)
+            }
+    }
+
+    fun dispose() {
+        compositeDisposable.dispose()
+    }
+
+    fun handleIceCandidateChange(iceCandidateState: IceCandidateState) {
+        when (iceCandidateState) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/8ce941c6c7cc4d13bad0e73c61419fa5d1ca7b0f/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/client/RtcClientMessageController.kt#L48"></span>Prefer using ‘if’ for binary conditions instead of ‘when’.</td>
    </tr>
  </tbody>
</table>